python_modules/dagster/dagster/core/definitions/job_definition.py

```diff
@@ -425,14 +425,23 @@ def partitions_def(self) -> Optional[PartitionsDefinition]:
 
         return mode.partitioned_config.partitions_def
 
-    def run_request_for_partition(self, partition_key: str, run_key: Optional[str]) -> RunRequest:
+    def run_request_for_partition(
+        self,
+        partition_key: str,
+        run_key: Optional[str],
+        tags: Optional[Dict[str, str]] = None,
+    ) -> RunRequest:
         partition_set = self.get_partition_set_def()
         if not partition_set:
             check.failed("Called run_request_for_partition on a non-partitioned job")
 
         partition = partition_set.get_partition(partition_key)
         run_config = partition_set.run_config_for_partition(partition)
-        tags = partition_set.tags_for_partition(partition)
+        if tags:
+            tags.update(partition_set.tags_for_partition(partition))
```

Comment:
I think that this will mutate the passed in dictionary, which might have unexpected consequences for the caller.   Could be safer to use this pattern instead:
```
run_request_tags = {**partition_set.tags_for_partition(partition), **tags}
```

Reply:
I think, in practice, it doesn't matter that much.  People don't tend to pass in / override dagster tags.

As a pattern, we tend to put the user-provided tags last, since those are the most confounding in terms of wondering why the thing you provided is not available as a tag.